### PR TITLE
fix: report when player state is error

### PR DESCRIPTION
**Pull request by pauldambra**
* let's capture an event when we show an error
* reduce the likelihood we'll show an error unnecessarily
* *fly-by improvements to the toolbar's event debugger cos i was using it to figure this out
* don't overreport some settings changes

Screenshot 2024-12-17 at 19 45 42

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 1]
```diff
@@ -1018,7 +1018,7 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
             cache.pausedMediaElements = values.endReached ? [] : playingElements
         },
         restartIframePlayback: () => {
-            cache.pausedMediaElements.forEach((el: HTMLMediaElement) => el.play())
+            cache.pausedMediaElements?.forEach((el: HTMLMediaElement) => el.play())
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:1021]: saw an error from trying to access this when it was undefined while debugging the rest of this

File: frontend/src/toolbar/debug/EventDebugMenu.tsx [hunk 2]
```diff
@@ -167,7 +189,13 @@ export const EventDebugMenu = (): JSX.Element => {
                 </div>
             </ToolbarMenu.Body>
             <ToolbarMenu.Footer noPadding>
-                <SettingsBar border="top" className="justify-end">
+                <SettingsBar border="top" className="justify-between">
+                    <SettingsMenu
```
Comment (pauldambra) [inline frontend/src/toolbar/debug/EventDebugMenu.tsx:193]: found myself needing this while figuring out the rest of this

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 3]
```diff
@@ -441,8 +441,6 @@ export const eventUsageLogic = kea<eventUsageLogicType>([
         reportRecordingsListPropertiesFetched: (loadTime: number) => ({ loadTime }),
         reportRecordingsListFilterAdded: (filterType: SessionRecordingFilterType) => ({ filterType }),
         reportRecordingPlayerSeekbarEventHovered: true,
-        reportRecordingPlayerSpeedChanged: (newSpeed: number) => ({ newSpeed }),
```
Comment (pauldambra) [inline frontend/src/lib/utils/eventUsageLogic.ts:444]: i hate this gigantic logic that just hides calling posthog capture

File: frontend/src/scenes/session-recordings/player/playerSettingsLogic.ts [hunk 4]
```diff
@@ -122,4 +123,13 @@ export const playerSettingsLogic = kea<playerSettingsLogicType>([
             (preferredSidebarStacking) => preferredSidebarStacking === SessionRecordingSidebarStacking.Vertical,
         ],
     }),
+
+    listeners({
+        setSpeed: ({ speed }) => {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/playerSettingsLogic.ts:128]: moved these captures closer to the source

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 5]
```diff
@@ -848,9 +842,10 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
         startScrub: () => {
             actions.stopAnimation()
         },
-        setSpeed: ({ speed }) => {
-            actions.reportRecordingPlayerSpeedChanged(speed)
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:852]: there are always three instances of the player logic, all three are listening to the player settings logic, so all three were reporting this making usage look higher than it is... luckily not an important metric

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 6]
```diff
@@ -1143,6 +1138,16 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
                 actions.reportMessageTooLargeWarningSeen(values.sessionRecordingId)
             }
         },
+        currentPlayerState: (prev, next) => {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:1141]: pretty sure we're setting this more than we need to, but this at least can report on that too high a number

File: frontend/src/scenes/session-recordings/player/PlayerMeta.tsx [hunk 7]
```diff
@@ -59,26 +59,6 @@ function URLOrScreen({ lastUrl }: { lastUrl: string | undefined }): JSX.Element
     )
 }
 
-function PlayerWarningsRow(): JSX.Element | null {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/PlayerMeta.tsx:62]: we don't set these anywhere any more. so we don't need this code

File: frontend/src/scenes/session-recordings/player/controller/PlayerController.tsx [hunk 8]
```diff
@@ -82,7 +82,7 @@ function ShowMouseTail(): JSX.Element {
 
     return (
         <SettingsToggle
-            title="Show mouse tail"
+            title="Show a tail following the cursor to make it easier to see"
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/controller/PlayerController.tsx:85]: various changes to ensure there are tooltips on buttons

File: frontend/src/scenes/session-recordings/player/controller/PlayerController.tsx [hunk 9]
```diff
@@ -125,39 +155,15 @@ function InspectDOM(): JSX.Element {
...
-                            active: timestampFormat === TimestampFormat.Relative,
-                        },
-                    ]}
-                    icon={<IconClock />}
-                    label={TimestampFormatToLabel[timestampFormat]}
-                />
-                <InspectDOM />
+                <SetTimeFormat />
             </div>
+            <InspectDOM />
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/controller/PlayerController.tsx:166]: and to separate the inspect dom button from the player buttons

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 10]
```diff
@@ -156,7 +151,8 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
         endBuffer: true,
         startScrub: true,
         endScrub: true,
-        setErrorPlayerState: (show: boolean) => ({ show }),
+        setPlayerError: (reason: string) => ({ reason }),
+        clearPlayerError: true,
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:155]: "show" didn't really mean show, so now we store the reason and have a `clear` action

**Review by marandaneto — APPROVED** [review]

File: frontend/src/scenes/session-recordings/player/SessionRecordingPlayer.tsx [hunk 11]
```diff
@@ -158,9 +155,6 @@ export function SessionRecordingPlayer(props: SessionRecordingPlayerProps): JSX.
         }
     )
 
-    const lessThanFiveMinutesOld = dayjs().diff(start, 'minute') <= 5
-    const cannotPlayback = snapshotsInvalid && lessThanFiveMinutesOld && !messageTooLargeWarnings
-
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/SessionRecordingPlayer.tsx:163]: giving this a better name

**Pull request merged**